src/annoylib.h

```diff
@@ -164,8 +148,14 @@ struct Angular {
       return random->flip();
   }
   static inline void create_split(const vector<node*>& nodes, int f, Randomness<T>* random, node* n) {
+    // Sample two random points from the set of nodes
+    // Calculate the hyperplane equidistant from them
+    int i = random->index(nodes.size());
+    int j = random->index(nodes.size());
```

Comment:
I fixed it later.

Yes you can do something like that too. I wonder if there are some cheap tricks in practice where you can compute statistics on subsamples, eg. sample 10 random points and pick the pair of points that gives the most even split. Sounds a bit more messy to implement though!